content/300-guides/400-migrate-to-prisma/03-migrate-from-mongoose.mdx

```diff
@@ -0,0 +1,1056 @@
+  published  Boolean
+  title      String
+}
+
+model users {
+  id      String        @id @default(auto()) @map("_id") @db.ObjectId
+  v       Int           @map("__v")
+  email   String        @unique(map: "email_1")
+  name    String
+  profile UsersProfile?
+}
+```
+
+The generated Prisma models represent the collections and are the foundation of your programmatic Prisma Client API which allows you to send queries to your database.
+
+### 2.4. Adjust the relationships
+
+MongoDB doesn't support relations between different collections. However, you can create references from one document to another using the `ObjectId` schema field or from one document to many using an array of `ObjectIds`. The reference will store id(s) of the related document(s). You can use the `populate()` method Mongoose provides to populate the reference with the data of the related document.
+
+Update the 1-n relationship between `Post` <-> `User` as follows:
+
+- Rename the existing `author` reference in the `Post` to `authorId` and add the `@map("author")` attribute
+- Add the `posts` relation in the `users` model
+
+<TabbedContent tabs={[<FileWithIcon text="diff" icon="file"/>, <FileWithIcon text="schema.prisma" icon="file"/>]}>
+
+<tab>
+
+```prisma
+type UsersProfile {
+  bio String
+}
+
+model categories {
+  id   String @id @default(auto()) @map("_id") @db.ObjectId
+  v    Int    @map("__v")
+  name String
+}
+
+model posts {
+  id         String   @id @default(auto()) @map("_id") @db.ObjectId
+  v          Int      @map("__v")
+  title      String
+  content    String
+  published  Boolean
+
+-  author     String   @db.ObjectId
++  author     users  @relation(fields: [authorId], references: [id])
++  authorId   String @map("author") @db.ObjectId
+
+  categories String[] @db.ObjectId
+}
+
+model users {
+  id      String        @id @default(auto()) @map("_id") @db.ObjectId
+  v       Int           @map("__v")
+  email   String        @unique(map: "email_1")
+  name    String
+  profile UsersProfile?
++  posts   posts[]
+}
+```
+
+</tab>
+
+<tab>
+
+```prisma
+type UsersProfile {
+  bio String
+}
+
+model categories {
+  id   String @id @default(auto()) @map("_id") @db.ObjectId
+  v    Int    @map("__v")
+  name String
+}
+
+model posts {
+  id        String  @id @default(auto()) @map("_id") @db.ObjectId
+  title     String
+  content   String
+  published Boolean
+  v         Int     @map("__v")
+
+  author   users  @relation(fields: [authorId], references: [id])
+  authorId String @map("author") @db.ObjectId
+
+  categories String[] @db.ObjectId
+}
+
+model users {
+  id      String        @id @default(auto()) @map("_id") @db.ObjectId
+  v       Int           @map("__v")
+  email   String        @unique(map: "email_1")
+  name    String
+  profile UsersProfile?
+  posts   posts[]
+}
+```
+
+</tab>
+
+</TabbedContent>
+
+Update the m-n between `Post` <-> `Category` references as follows:
+
+- Update the `categories` field data type in the `posts` from `String[]`
```

Comment:
What you actually do here is rename the `categories` to `categoryId` and map it via `@map("categories")`, then add a new relation field `categories categories[] @relation(...)`.